webtau-junit5-examples-kotlin/src/test/kotlin/com/example/tests/junit5/CustomerCrudSeparatedKotlinTest.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.tests.junit5
+
+import com.twosigma.webtau.WebTauDsl.*
+import com.twosigma.webtau.http.validation.HttpResponseValidatorWithReturn
+import com.twosigma.webtau.junit5.WebTau
+
+import org.junit.jupiter.api.*
+
+@WebTau
+@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
+@DisplayName("customer")
+class CustomerCrudSeparatedKotlinTest {
+    @Test
+    @Order(1)
+    fun `read customer record`() {
+        http.get("/customers/$id") { _, body ->
+            body.should(equal(customerPayload))
+        }
+    }
+
+    @Test
+    @Order(2)
+    fun `update customer record`() {
+        http.put("/customers/$id", changedCustomerPayload) { _, body ->
+            body.should(equal(changedCustomerPayload))
+        }
+
+        http.get("/customers/$id") { _, body ->
+            body.should(equal(changedCustomerPayload))
+        }
+    }
+
+    @Test
+    @Order(3)
+    fun `delete customer`() {
+       http.delete("/customers/$id") { header, _ ->
+           header.statusCode().should(equal(204))
+       }
+
+        http.get("/customers/$id") { header, _ ->
+            header.statusCode().should(equal(404))
+        }
+    }
+
+    companion object {
+        private val customerPayload = mapOf(
+            "firstName" to "FN",
+            "lastName" to "LN"
+        )
+        private val changedCustomerPayload = mapOf(
+            "lastName" to "NLN"
+        )
+        private val id by lazy {
+            val id: Int = http.post("/customers", customerPayload, HttpResponseValidatorWithReturn { _, body ->
```

Comment:
yeah, this looks a bit weird.

Reply:
It's not uncommon in Kotlin code which uses java libs sadly but we can provide Kotlin specific versions of the http methods which won't need this.

Reply:
will Kotlin specific versions need to provide all the overrides? or is there some smart extension mechanism?

Reply:
It has an extension mechanism.  This is something I've been playing with for should:
```kotlin
infix fun Any.shouldNot(valueMatcher: ValueMatcher) = actual(this).shouldNot(valueMatcher))
```
Which would allow me to do:
```
val id: Int = http.post("/customers", customerPayload, HttpResponseValidatorWithReturn { _, body ->
                body.get("id")
            })

id shouldNot equal(0)
```

I'm pretty sure I can make the http methods work nicely and feel Kotlin native without tonnes of work but that'll be for a separate PR.

Reply:
oh neat. separate PR sounds good.